### PR TITLE
EWPP-2997: Drupal 10 compatibility.

### PR DESCRIPTION
**Pull request by brummbar**

### CODE REVIEW

**Review by drishu — DISMISSED** [review]

File: composer.json [hunk 1]
```diff
@@ -75,11 +61,8 @@
             }
         },
         "_readme": [
-            "Use master branch of drupal/drupal-extension to replace fabpot/goutte with behat/mink-browserkit-driver.",
             "Explicit minimum version requirement of symfony/phpunit-bridge to replace drupal/core-dev testing classes and traits.",
```
Comment (22Alexandra) [inline composer.json]: We should remove this readme entry as the dependency has been removed.

File: composer.json [hunk 2]
```diff
@@ -8,32 +8,18 @@
...
-        "composer/installers": "~1.5",
-        "drupal/core-composer-scaffold": "^9.4",
-        "drupal/config_devel": "~1.2",
-        "drupal/drupal-extension": "dev-master as 4.0",
+        "composer/installers": "^1.11",
+        "drupal/core-composer-scaffold": "^9.4 || ^10",
+        "drupal/core-dev": "^9.4 || ^10",
         "drush/drush": "^11.1",
-        "guzzlehttp/guzzle": "^7.0",
         "mikey179/vfsstream": "^1.6.10",
```
Comment (22Alexandra) [inline composer.json]: We can remove this dependency and its readme entry.

File: tests/modules/media_avportal_mock/src/AvPortalClientMiddleware.php [hunk 3]
```diff
@@ -36,17 +37,27 @@ class AvPortalClientMiddleware {
...
    * @param \Symfony\Component\EventDispatcher\EventDispatcherInterface $eventDispatcher
    *   The event dispatcher.
+   * @param \Drupal\Core\Extension\ModuleExtensionList|null $moduleExtensionList
+   *   The module extension list.
    */
-  public function __construct(ConfigFactoryInterface $configFactory, EventDispatcherInterface $eventDispatcher) {
+  public function __construct(ConfigFactoryInterface $configFactory, EventDispatcherInterface $eventDispatcher, ModuleExtensionList $moduleExtensionList = NULL) {
     $this->configFactory = $configFactory;
     $this->eventDispatcher = $eventDispatcher;
+    $this->moduleExtensionList = $moduleExtensionList ?? \Drupal::service('extension.list.module');
```
Comment (22Alexandra) [inline tests/modules/media_avportal_mock/src/AvPortalClientMiddleware.php:60]: When is this needed? The service seems to be always available.

**Review by 22Alexandra — COMMENTED** [review]
Minor comments.

File: tests/modules/media_avportal_mock/src/AvPortalClientMiddleware.php (hunk 3 above)
Comment (brummbar) [reply tests/modules/media_avportal_mock/src/AvPortalClientMiddleware.php:60]: It's to add a layer of backwards compatibility for users that extended this service.

**Review by 22Alexandra — APPROVED** [review]

**Review by drishu — APPROVED** [review]

**Pull request merged**